src/sparsezoo/analysis/Analysis.py

```diff
@@ -0,0 +1,221 @@
+    sparsity: float = Field(description="Proportion of zeros in this node's parameter")
+    four_block_sparsity: float = Field(
+        description="Proportion of zero blocks in this node's parameter"
+    )
+    param_size: int = Field(description="Size of this node's parameter")
+    num_sparse_values: int = Field(
+        description="Number of zeros in this node's parameter"
+    )
+    num_four_blocks: int = Field(
+        description="Number of four blocks in this node's parameter"
+    )
+    num_sparse_four_blocks: int = Field(
+        description="Number of four blocks with all zeros in this node's parameter"
+    )
+    quantized_layer: bool = Field(description="Does this node have quantized weights")
+    zero_point: int = Field(
+        description="Node zero point for quantization, default zero"
+    )
+    dtype: Optional[str] = Field(
+        description="This node's parameter's data type", default=None
+    )
+
+
+class ModelAnalysis(BaseModel):
+    """
+    Pydantic model for analysis of a model
+    """
+
+    layer_counts: Dict[str, int] = Field(description="Overview of layers", default={})
+    op_counts: Dict[str, int] = Field(
+        description="Overview of operations (nodes that are not layers)", default={}
+    )
+    num_nodes: int = Field(description="Total number of nodes")
+    num_layers: int = Field(description="Total number of layers")
+    num_operations: int = Field(
+        description="Total number of operations (nodes that are not layers)"
+    )
+    average_sparsity: float = Field(
+        description="Average sparsity across all trainable parameters "
+        "(excluding biases)"
+    )
+    average_four_block_sparsity: float = Field(
+        description="Average sparsity four block sparsity across all "
+        "trainable parameters (excluding biases)"
+    )
+    num_sparse_layers: int = Field(description="Number of layers with any sparsity")
+    num_quantized_layers: int = Field(description="Number of quantized layers")
+    num_parameters: int = Field(
+        description="Total number of all parameters (excluding biases)"
+    )
+    num_sparse_parameters: int = Field(
+        description="Total number of all sparsified parameters (excluding biases)"
+    )
+    num_four_blocks: int = Field(
+        description="Total number of all four blocks (excluding biases)"
+    )
+    num_sparse_four_blocks: int = Field(
+        description="Total number of all sparsified four blocks (excluding biases)"
+    )
+
+    nodes: List[NodeAnalysis] = Field(description="List of node analyses", default=[])
+
+    @classmethod
+    def from_onnx_model(cls, onnx_file_path: str):
+        """
+        Class constructor for Model Analysis
+
+        :param cls: class being constructed
+        :param onnx_file_path: path to onnx file being analyzed
+        :return: instance of ModelAnalysis class
+        """
+        model_onnx = onnx.load(onnx_file_path)
+
+        param_prunable_nodes = list(
```

Comment:
for readability and since we are casting to a list anyways, I'd suggest switching this to a list comprehension with an `if` for filtering